crates/zizmor/src/audit/bot_conditions.rs

```diff
@@ -200,11 +228,245 @@ impl BotConditions {
+                                && lit.as_str().ends_with("[bot]"))
+                                || (SPOOFABLE_ACTOR_ID_CONTEXTS.iter().any(|x| x.matches(ctx))
+                                    && BOT_ACTOR_IDS.contains(&lit.as_str().as_ref()))
+                            {
+                                fragments.push(ctx.as_str().to_string());
+                            }
+                        }
+                        _ => {}
+                    }
+                }
+
+                fragments.extend(Self::find_spoofable_actor_fragments(lhs));
+                fragments.extend(Self::find_spoofable_actor_fragments(rhs));
+            }
+            Expr::UnOp { expr, .. } => {
+                fragments.extend(Self::find_spoofable_actor_fragments(expr));
+            }
+            _ => {}
+        }
+
+        fragments
+    }
+
+    /// Create a fix that replaces spoofable actor contexts with github.event.pull_request.user.login
+    fn create_replace_actor_fix<'doc>(step: &Step<'doc>) -> Option<Fix<'doc>> {
+        // Only emit a patch if the step/job has an `if` key
+        if let Some(If::Expr(expr)) = &step.r#if {
+            // Try to parse as curly expression first, otherwise use the raw string
+            let bare_expr = match ExplicitExpr::from_curly(expr) {
+                Some(raw_expr) => raw_expr.as_bare().to_string(),
+                None => expr.to_string(),
+            };
+
+            let Ok(parsed_expr) = Expr::parse(&bare_expr) else {
+                return None;
+            };
+
+            // Find all spoofable actor fragments in the expression
+            let fragments = Self::find_spoofable_actor_fragments(&parsed_expr);
+
+            if fragments.is_empty() {
+                return None;
+            }
+
+            // Create patches for each unique fragment
+            let mut patches = vec![];
+            let mut seen_fragments = std::collections::HashSet::new();
+
+            for fragment in fragments {
+                if seen_fragments.insert(fragment.clone()) {
+                    // Replace the fragment with the secure alternative
+                    let replacement = if SPOOFABLE_ACTOR_NAME_CONTEXTS.iter().any(|pattern| {
+                        // Parse the fragment to check if it matches our patterns
+                        if let Ok(test_expr) = Expr::parse(&fragment) {
+                            if let Expr::Context(ctx) = test_expr {
+                                pattern.matches(&ctx)
+                            } else {
+                                false
+                            }
+                        } else {
+                            false
+                        }
+                    }) {
+                        "github.event.pull_request.user.login"
+                    } else {
+                        // This is an actor_id context, replace with user.id
+                        "github.event.pull_request.user.id"
```

Comment:
For full generality I _think_ we need to handle different event types here. Specifically, I think we need to look at the `on` trigger block and see which event(s) are used to suggest the right context here (e.g. `github.event.comment.user.login`). Otherwise we'll end up suggesting a mismatched context here 🙂